sdk/storage/inc/blobs/blob_service_client.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "blob_client_options.hpp"
+#include "common/storage_credential.hpp"
+#include "common/storage_url_builder.hpp"
+#include "internal/protocol/blob_rest_client.hpp"
+
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Storage { namespace Blobs {
+
+  class BlobServiceClient {
+  public:
+    // connection string
+    static BlobServiceClient CreateFromConnectionString(
+        const std::string& connectionString,
+        const BlobServiceClientOptions& options = BlobServiceClientOptions());
+
+    // shared key auth
+    explicit BlobServiceClient(
+        const std::string& serviceUri,
+        std::shared_ptr<SharedKeyCredential> credential,
+        const BlobServiceClientOptions& options = BlobServiceClientOptions());
+
+    // token auth
+    explicit BlobServiceClient(
+        const std::string& serviceUri,
+        std::shared_ptr<TokenCredential> credential,
+        const BlobServiceClientOptions& options = BlobServiceClientOptions());
+
+    // anonymous/SAS/customized pipeline auth
+    explicit BlobServiceClient(
+        const std::string& serviceUri,
+        const BlobServiceClientOptions& options = BlobServiceClientOptions());
+
+    ListContainersSegment ListBlobContainersSegment(
+        const ListBlobContainersOptions& options = ListBlobContainersOptions()) const;
+
+    UserDelegationKey GetUserDelegationKey(
+        const std::string& expiresOn,
```

Comment:
nit: What is the indentation guide here. I'm seeing 2/4 spaces mixed.

Reply:
i don't know. it's formatted by clang-format, and I think it's okay.